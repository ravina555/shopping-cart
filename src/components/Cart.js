import React,{useEffect,useState} from 'react';
import ShoppingCard from './ShoppingCard';
import '../App.css';



const Cart =()=>{
    const [Products,setProducts]=useState([]);


    useEffect(()=>{

        fetch('https://fakestoreapi.com/products')
                .then(res=>res.json())
                .then(json=>{
                    setProducts(json);
        });            
        
    },[])

    return(
        <div className='cartBg'>
            <div className='Products'>
            <ShoppingCard Products={Products}/>
            </div>           
        </div>
    )
}

export default Cart;