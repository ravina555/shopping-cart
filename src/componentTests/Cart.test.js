import Cart from '../components/Cart';
import Enzyme,{shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new Adapter()})
describe('should render Cart component',()=>{
    it('should match snapshot',()=>{
        let cart = renderer.create(<Cart/>);
        expect(cart).toMatchSnapshot();
    });
    it('should have children component',()=>{
        let cartInstance = shallow(<Cart/>);
        expect(cartInstance.props().children).toBeDefined();
        expect(cartInstance.props().className).toBe('cartBg');
    });
    xit('should fetch the product data',done=>{
        // const wrapper = mount(<Cart/>);
        // console.log(wrapper)
      
    })
})