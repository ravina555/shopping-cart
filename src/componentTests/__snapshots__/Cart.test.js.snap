// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Cart component should match snapshot 1`] = `
<div
  className="cartBg"
>
  <div
    className="Products"
  />
</div>
`;
